src/rules/no-large-snapshots.js

```diff
@@ -10,8 +10,29 @@ const reportOnViolation = (context, node) => {
   const startLine = node.loc.start.line;
   const endLine = node.loc.end.line;
   const lineCount = endLine - startLine;
+  const whitelistedSnapshots =
+    context.options &&
+    context.options[0] &&
+    context.options[0].whitelistedSnapshots;
+
+  let isWhitelisted = false;
+  if (whitelistedSnapshots) {
+    const fileName = path.relative(process.cwd(), context.getFilename());
```

Comment:
this seems risky and potentially surprising - could we ask people to provide absolute paths instead? They'd have to use `.eslintrc.js` but I think that's fine. It's only js files mentioned in the eslint@6 migration guide: https://eslint.org/docs/6.0.0/user-guide/migrating-to-6.0.0. They'd then do `require.resolve('./path-to-my-file')` or something like it.

The reason I want thit is that I'd expect the relative path provided in config to be relative to the configuration file, not to `cwd`

Reply:
@SimenB absolsute paths work because we can’t check that into source control.  I’m good with doing it relative to the config file, is there some mechanism I can use to get that directory?

Reply:
People can use js to calculate absolute paths within the config

Reply:
@SimenB what do you think of a special property for the root directory, so the root dir can be specified once, rather than in each file name?  so something like this:

```javascript
{
  whitelistedSnapshots: {
    "__root":"/absolute/path/to/root", //defaults to process.cwd()
    "relative/path/to/snapshot.snap": ["test1", "test2"],
    "relative/path/to/snapshot2.snap": ["test1", "test3"],
  }

```

That would be much better logistically for teams like us with hundreds and hundreds of large snapshots that we need to grandfather in.

Reply:
We have `<rootDir>` in Jest and IMO the UX is mediocre. I don't think it's worth it to introduce a concept that people need to understand over using standard `path` mechanics.
Aside from the lack of support for JSON config, which may or may not be considered a big deal, it should be pretty easy to do something like
```js
const base = path.resolve(__dirname, 'src');
const whitelist = [path.resolve(base, 'file1.js'), path.resolve(base, 'file2.js')];
```
Could even introduce another function that applies the `base` argument already.

Reply:
I don't think banning relative paths outright is strictly necessary, but I also can't think of a good base directory (cwd might vary, config file location makes no sense if it's in a `config` directory) so idk if we have another option

Reply:
@jeysal the reason for doing this is because we have hundreds maybe even thousands of large snapshots that we need to whitelist so actually it would be quite painful.

Reply:
As soon as you make a helper function, possibly even one that takes multiple arguments and resolves them all from the base (`snapshots('1.js', '2.js')`), I don't see how it is more painful though?
On a side note, maybe for your scale you might want to make large snapshots follow a pattern that can be matched on or something?

Reply:
@jeysal @SimenB ok i went ahead and got rid of the `process.cwd()` and updated the docs as well.  it will now assume an absolute path.